### PR TITLE
End to end tests improvements

### LINKED ISSUE #95: Move end-to-end tests in test folder
Curently the `end-to-end` tests sources are positioned in the `src` folder. They also rely on interfaces placed in the `examples` folder.
This is not conform to Rust [best practices](https://doc.rust-lang.org/cargo/guide/project-layout.html).
All the `end-to-end` sources and dependencies should be moved to the `tests` folder.

### PR DESCRIPTION
**Pull request by sorru94**
Improves the end to end tests.
With this PR tests are performed for:
- Individual datastreams, direction Device -> Server
- Individual datastreams, direction Server -> Device
- Aggregate datastreams, direction Device -> Server
- Aggregate datastreams, direction Server -> Device
- Individual properties, direction Device -> Server
- Individual properties, direction Server -> Device

All the `AstarteTypes` have been tested. 
The only exceptions are `BinaryBlob`, `BinaryBlobArray`, `DateTime` and `DateTimeArray` for the datastream aggregates in the direction Server -> Device. This is due to two `bugs` in Astarte ([bug1](https://github.com/astarte-platform/astarte/issues/753) and [bug2](https://github.com/astarte-platform/astarte/issues/754)).
Closes #95

### CODE REVIEW

File: .reuse/dep5 [hunk 1]
```diff
@@ -5,3 +5,11 @@ Source: https://github.com/astarte-platform/astarte-device-sdk-rust
...
 Copyright: SECO Mind Srl
 License: CC0-1.0
+
+Files: tests/e2etest/interfaces/*
+Copyright: SECO Mind Srl
+License: CC0-1.0
+
+Files: tests/e2etest/mock_data/*
+Copyright: SECO Mind Srl
+License: CC0-1.0
```
Comment (harlem88) [inline .reuse/dep5:15]: Does the mock_data folder still exist?

File: tests/e2etest/mock_data_aggregate.rs [hunk 2]
```diff
@@ -0,0 +1,376 @@
+/*
+ * This file is part of Astarte.
+ *
+ * Copyright 2022 SECO Mind Srl
```
Comment (harlem88) [inline tests/e2etest/mock_data_aggregate.rs:4]: Change all license header year to 2023.

**Review by harlem88 — CHANGES_REQUESTED** [review]

File: .reuse/dep5 (hunk 1 above)
Comment (sorru94) [reply .reuse/dep5:15]: Nope

File: tests/e2etest/mock_data_aggregate.rs [hunk 3]
```diff
@@ -0,0 +1,376 @@
...
+            Some(("integer_endpoint".to_string(), 20)),
+            Some(("boolean_endpoint".to_string(), true)),
+            Some(("longinteger_endpoint".to_string(), 45993543534)),
+            Some(("string_endpoint".to_string(), "string for test".to_string())),
+            // Some((
+            //     "binaryblob_endpoint".to_string(),
+            //     base64::engine::general_purpose::STANDARD
+            //         .decode("d29ybGQ=")
+            //         .unwrap(),
+            // )),
```
Comment (FrancescoVaiani) [inline tests/e2etest/mock_data_aggregate.rs:105]: Please do not commit commented code. Leave a TODO comment to make it evident something should be there.

File: tests/e2etest/mock_data_aggregate.rs [hunk 4]
```diff
@@ -0,0 +1,376 @@
...
+            None,
+            // TODO workaround for bug in Astarte:
+            // Some((
+            //     "datetime_endpoint".to_string(),
+            //     DateTime::<Utc>::from(
+            //         DateTime::parse_from_rfc3339("2022-08-29T17:46:48.000Z").unwrap(),
+            //     ),
+            // )),
```
Comment (FrancescoVaiani) [inline tests/e2etest/mock_data_aggregate.rs:113]: Same as above

File: tests/e2etest/mock_data_aggregate.rs [hunk 5]
```diff
@@ -0,0 +1,376 @@
...
+            //     "binaryblobarray_endpoint".to_string(),
+            //     Vec::from([
+            //         base64::engine::general_purpose::STANDARD
+            //             .decode("aGVsbG8=")
+            //             .unwrap(),
+            //         base64::engine::general_purpose::STANDARD
+            //             .decode("aGVsbG8=")
+            //             .unwrap(),
+            //     ]),
+            // )),
```
Comment (FrancescoVaiani) [inline tests/e2etest/mock_data_aggregate.rs:145]: Same as above

File: tests/e2etest/mock_data_aggregate.rs [hunk 6]
```diff
@@ -0,0 +1,376 @@
...
+            //             DateTime::parse_from_rfc3339("2022-06-29T17:46:48.000Z").unwrap(),
+            //         ),
+            //         DateTime::<Utc>::from(
+            //             DateTime::parse_from_rfc3339("2022-11-29T17:46:49.000Z").unwrap(),
+            //         ),
+            //         DateTime::<Utc>::from(
+            //             DateTime::parse_from_rfc3339("2022-10-29T17:46:50.000Z").unwrap(),
+            //         ),
+            //     ]),
+            // )),
```
Comment (FrancescoVaiani) [inline tests/e2etest/mock_data_aggregate.rs:161]: Same as above

File: tests/e2etest/mock_data_aggregate.rs (hunk 3 above)
Comment (sorru94) [reply tests/e2etest/mock_data_aggregate.rs:105]: Let's do that I open an issue regarding those missing tests.

**Review by harlem88 — APPROVED** [review]

**Review by FrancescoVaiani — APPROVED** [review]

**Review by harlem88 — APPROVED** [review]

**Pull request merged**